// react-native-qrcode-mask/lib/commonjs/Edge.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * Created by nghinv on Thu Jul 15 2021
 * Copyright (c) 2021 nghinv@lumi.biz
 */
function Edge(props) {
  const {
    edgeWidth,
    edgeHeight,
    edgeColor,
    edgeBorderWidth,
    type
  } = props;
  const [type1, type2] = type.split('_');
  const edgeStyle = {
    [type1.toLocaleLowerCase()]: 0,
    [type2.toLocaleLowerCase()]: 0,
    [`border${type1}Width`]: edgeBorderWidth,
    [`border${type2}Width`]: edgeBorderWidth
  };
  return /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: [styles.container, {
      width: edgeWidth,
      height: edgeHeight,
      borderColor: edgeColor
    }, edgeStyle]
  });
}

const styles = _reactNative.StyleSheet.create({
  container: {
    position: 'absolute'
  }
});

var _default = /*#__PURE__*/_react.default.memo(Edge);

exports.default = _default;
